ee/localserver/krypto-ec-middleware.go

```diff
@@ -459,3 +477,197 @@ func (bhr *bufferedHttpResponse) WriteHeader(code int) {
 func (bhr *bufferedHttpResponse) Bytes() []byte {
 	return bhr.buf.Bytes()
 }
+
+func (e *kryptoEcMiddleware) presenceDetectionCallback(challengeBox *challenge.OuterChallenge) {
+	// extract cmd req from challenge box
+	ctx, cancel := context.WithTimeout(context.Background(), presencedetection.DetectionTimeout)
+	defer cancel()
+
+	var cmdReq v2CmdRequestType
+	if err := json.Unmarshal(challengeBox.RequestData(), &cmdReq); err != nil {
+		e.slogger.ErrorContext(ctx,
+			"error unmarshaling cmd request",
+			"err", err,
+		)
+		return
+	}
+
+	// figure out if we need to do presence detection
+	detectionIntervalStr, ok := cmdReq.Headers[kolidePresenceDetectionIntervalHeaderKey]
+	if !ok || len(detectionIntervalStr) == 0 {
+		return
+	}
+
+	// no presence detection requested
+	if len(detectionIntervalStr) == 0 || detectionIntervalStr[0] == "" {
```

Comment:
Is there a reason this isn't combined with line 497 above? It looks like the same len check in both of these `if`s

Reply:
fixed